src/iterator.hpp

```diff
@@ -52,7 +53,7 @@ template <typename T> class iterator_base {
     // since the iterator doesn't keep track of the root object provided,
     // but only the beginning of the key path, we keep this here so that we
     // can later provide the accurate full key path.
-    std::vector<std::string> path_;
+    memory::vector<memory::string> path_;
```

Comment:
I don't think this should use the monotonic buffer. You can possibly create many iterators, and their memory will never be reclaimed.

Reply:
Alright, that's fair enough.